python/ngen_conf/src/ngen/config/init_config/noahowp.py

```diff
@@ -270,4 +271,27 @@ class Config(core.Base.Config):
         }
 
 
+def _warn_if_soil_or_veg_type_is_water_but_not_both(
+    parameters: "Parameters", structure: "Structure"
+):
+    if parameters.soil_class_name not in ("STAS", "STAS-RUC"):
+        return
+    SOIL_TYPE_WATER = 14
+    soil_type_is_water = structure.isltyp == SOIL_TYPE_WATER
+
+    if parameters.veg_class_name == "USGS":
+        VEG_USGS_WATER = 16
+        veg_type_is_water = structure.vegtyp == VEG_USGS_WATER
+    elif parameters.veg_class_name == "MODIFIED_IGBP_MODIS_NOAH":
+        VEG_MODIS_WATER = 17
+        veg_type_is_water = structure.vegtyp == VEG_MODIS_WATER
+    else:
+        return
+
+    if (veg_type_is_water) ^ (soil_type_is_water):
```

Comment:
This only works "correctly" when the operands are `bool` types, which by the code above ensures they are, so you the logical and bitwise exclusive or are equivalent.  It can be a bit of a foot gun if this function is modified, or copied/used in the future for different validation without understanding this, but I don't see much need to change it now.  Perhaps a quick comment to the above and the use of bitwise XOR to save someone in the future some possible headache?

Reply:
Yeah, I think that is fair. Thanks for mentioning it! Just added a comment.